File: packages/shared/src/couriers.js
export const COURIERS = {
  starken: {
    id: 'starken',
    name: 'Starken',
    color: '#e11d48',
    getTrackingUrl: (code) =>
      code ? `https://www.starken.cl/seguimiento?codigo=${encodeURIComponent(code.trim())}` : 'https://www.starken.cl/seguimiento',
  },
  chilexpress: {
    id: 'chilexpress',
    name: 'Chilexpress',
    color: '#eab308',
    getTrackingUrl: (code) =>
      code ? `https://www.chilexpress.cl/tracking-envio?ot=${encodeURIComponent(code.trim())}` : 'https://www.chilexpress.cl/tracking-envio',
  },
  blue_express: {
    id: 'blue_express',
    name: 'Blue Express',
    color: '#2563eb',
    getTrackingUrl: (code) =>
      code ? `https://www.bluex.cl/seguimiento/?tracking=${encodeURIComponent(code.trim())}` : 'https://www.bluex.cl/seguimiento/',
  },
  correos_chile: {
    id: 'correos_chile',
    name: 'Correos de Chile',
    color: '#dc2626',
    getTrackingUrl: (code) =>
      code ? `https://www.correos.cl/seguimiento-en-linea?envio=${encodeURIComponent(code.trim())}` : 'https://www.correos.cl/seguimiento-en-linea',
  },
  pullman: {
    id: 'pullman',
    name: 'Pullman Cargo',
    color: '#0284c7',
    getTrackingUrl: () => 'https://www.pullmancargo.cl/',
  },
};

export const COURIER_LABELS = {
  starken: 'Starken',
  chilexpress: 'Chilexpress',
  blue_express: 'Blue Express',
  correos_chile: 'Correos de Chile',
  pullman: 'Pullman Cargo',
};

export const getCourierName = (courierId) => {
  if (!courierId) return 'Courier no asignado';
  const clean = String(courierId).toLowerCase().trim();
  return COURIER_LABELS[clean] || courierId;
};

export const getTrackingUrl = (courierId, trackingCode) => {
  if (!courierId) return null;
  const clean = String(courierId).toLowerCase().trim();
  const courier = COURIERS[clean];
  if (courier && courier.getTrackingUrl) {
    return courier.getTrackingUrl(trackingCode);
  }
  return null;
};

File: apps/api/app/core/couriers.py
import urllib.parse

COURIER_INFO = {
    "starken": {
        "name": "Starken",
        "url_template": "https://www.starken.cl/seguimiento?codigo={code}",
        "default_url": "https://www.starken.cl/seguimiento",
    },
    "chilexpress": {
        "name": "Chilexpress",
        "url_template": "https://www.chilexpress.cl/tracking-envio?ot={code}",
        "default_url": "https://www.chilexpress.cl/tracking-envio",
    },
    "blue_express": {
        "name": "Blue Express",
        "url_template": "https://www.bluex.cl/seguimiento/?tracking={code}",
        "default_url": "https://www.bluex.cl/seguimiento/",
    },
    "correos_chile": {
        "name": "Correos de Chile",
        "url_template": "https://www.correos.cl/seguimiento-en-linea?envio={code}",
        "default_url": "https://www.correos.cl/seguimiento-en-linea",
    },
    "pullman": {
        "name": "Pullman Cargo",
        "url_template": "https://www.pullmancargo.cl/",
        "default_url": "https://www.pullmancargo.cl/",
    },
}


def get_courier_name(courier_id: str) -> str:
    if not courier_id:
        return "Courier"
    clean = str(courier_id).strip().lower()
    return COURIER_INFO.get(clean, {}).get("name", courier_id.title())


def get_tracking_url(courier_id: str, tracking_code: str) -> str:
    if not courier_id:
        return ""
    clean = str(courier_id).strip().lower()
    info = COURIER_INFO.get(clean)
    if not info:
        return ""
    if tracking_code:
        quoted = urllib.parse.quote(str(tracking_code).strip())
        return info["url_template"].format(code=quoted)
    return info["default_url"]
